### PR TITLE
Add static analysis tools

### PR DESCRIPTION
**Pull request by rfernandes**
* Add targets for cppcheck and clang-tidy if they are available.
* Rework some of the includes as system includes.
* Turn make verbose option off (use VERBOSE=1 make when required).

### CODE REVIEW

**Comment by rfernandes** [comment]
The first of many PRs, enjoy

File: CMakeLists.txt [hunk 1]
```diff
@@ -173,6 +174,31 @@ endif()
...
+  add_custom_target(cppcheck COMMAND ${CPPCHECK}
+        --std=c++14
+        -j ${CORE_COUNT}
+        --quiet
+        --project=${PROJECT_BINARY_DIR}/compile_commands.json)
+endif()
+
+
+# Add clang-tidy target if tool is available
+find_program(CLANG_TIDY run-clang-tidy.py "/usr/share/clang/")
```
Comment (marayl) [inline CMakeLists.txt]: Do we need to consider `SWIRLY_TOOLSET` here?

**Comment by marayl** [comment]
Travis-CI failure is not caused by this PR. I will take a look tomorrow.

File: CMakeLists.txt (hunk 1 above)
Comment (rfernandes) [reply CMakeLists.txt]: I didn't have to set SWIRLY_TOOLSET at all to build in up to date Arch, but I'll add it to find_program as a HINTS entry

File: CMakeLists.txt [hunk 2]
```diff
@@ -173,6 +174,31 @@ endif()
...
+  add_custom_target(cppcheck COMMAND ${CPPCHECK}
+        --std=c++14
+        -j ${CORE_COUNT}
+        --quiet
+        --project=${PROJECT_BINARY_DIR}/compile_commands.json)
+endif()
+
+
+# Add clang-tidy target if tool is available
+find_program(CLANG_TIDY run-clang-tidy.py HINTS "/usr/share/clang/" ${SWIRLY_TOOLSET}/share/clang)
```
Comment (marayl) [inline CMakeLists.txt]: Quote `${SWIRLY_TOOLSET}/share/clang` in case in contains spaces.

**Pull request merged**